rfcs/20220815-v3.0-API-cleanup/README.md

```diff
@@ -0,0 +1,92 @@
+# RFC: v3.0 API cleanup
+
+## Introduction
+Major v3 oneDNN release is coming [soon](https://github.com/oneapi-src/oneDNN/milestone/16).
+This is a perfect opportunity to provide API improvements.
+To decrease maintenance pressure on the library, outdated APIs will be removed from the library which affects user integration code of oneDNN.
+oneDNN v3.0 will not be API/ABI backward compatible with oneDNN v2.x.
+Hence, user will need to modify their code to use oneDNN v3.0.
+Below one may find a comprehensive list of targeted changes:
+
+## List of changes
+Disclaimer: header files are copied in this folder are from oneDNN v2.7 to preserve lines to refer to exact spots.
+
+* Build:
+    * Build time option `DNNL_USE_RT_OBJECTS_IN_PRIMITIVE_CACHE` and all code supporting the `false` value will be removed.
+
+* C/C++ types:
+    * Eltwise algorithm types:
+        - [`soft_relu`](dnnl.hpp#L563) will be removed. [`soft_relu_v2`](dnnl.hpp#L565) will be renamed into `soft_relu`.
+            * End user must specify `alpha` value in a correspondent call.
+        - [`logsigmoid`](dnnl.hpp#L567) will be removed. It can be used as current [`soft_relu_v2`](dnnl.hpp#L565) with `alpha` equal to `-1`.
```

Comment:
Hi @kawakami-k, just a heads up for this one: I've noticed that aarch64 eltwise injector lacks soft_relu_v2 support. Since we are cleaning up API for v1 and logsigmoid, aarch64 backend will not have an actual support for that. If these are important algorithms for any of your customers, could you, please, plan an implementation for it, please?

Ho @nSircombe, same applies for ACL backend. I don't know exact constraints but if ACL soft relu doesn't support alpha, v2 algorithm should be used with a limitation of alpha equal to -1 instead of current v1. Could you check it for me, please?

Thank you.